third_party/cares/config_linux/ares_config.h

```diff
@@ -505,6 +505,30 @@
 # define _DARWIN_USE_64_BIT_INODE 1
 #endif
 
+#ifdef GPR_BACKWARDS_COMPATIBILITY_MODE
+  /* Redefine the fd_set macros for glibc < 2.15 support. */
```

Comment:
Maybe quickly add the explanation that the new glibc 2.15 behavior is to call __fdelt_chk which was introduced then.